tests/unit/storage/data/external/GcpService.spec.js

```diff
@@ -86,73 +87,123 @@ const operations = [
     },
 ];
 
+// Reusable cleanup function
+async function cleanupServer(httpServer, sockets) {
+    if (httpServer) {
+        sockets.forEach(socket => {
+            if (!socket.destroyed) {
+                socket.destroy();
+            }
+        });
+        
+        await new Promise((resolve) => {
+            const timeout = setTimeout(() => {
+                resolve();
+            }, 3000);
+            
+            httpServer.close(() => {
+                clearTimeout(timeout);
+                resolve();
+            });
+        });
+    }
+}
+
 describe('GcpService request behavior', () => {
     jest.setTimeout(120000);
     let httpServer;
     let client;
+    let sockets = [];
 
     beforeAll(done => {
         client = new GCP({
-            endpoint: `http://${host}`,
-            maxRetries: 0,
-            s3ForcePathStyle: false,
-            accessKeyId,
-            secretAccessKey,
+            s3Params: {
+                endpoint: `http://${host}`,
+                maxAttempts: 1,
+                forcePathStyle: false,
+                region: 'us-east-1', // Required by AWS SDK v3
```

Comment:
```suggestion
                region: 'us-east-1',
```

This comment make sense because you are switching the AWS SDK but in 3 months, this comment will not make sense anymore